app/src/main/java/com/cortado/mafkir/ui/EditFragment.kt

```diff
@@ -46,28 +57,208 @@ class EditFragment : DaggerFragment() {
     override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
         super.onViewCreated(view, savedInstanceState)
 
-        binding.mafkirContact = args.mafkirContact
-        binding.timeInterval =
-            timeConverter.millisToDays(args.mafkirContact.interactionIntervalMillis).toString()
+        setHasOptionsMenu(true)
+
+        setupBinding()
+
+        setupWeekdaysToggleGroup()
+
+        setupInterval()
+
+        setupIntervalType()
 
         setupViewModel()
 
-        binding.clickListener = View.OnClickListener {
-            mafkirContactViewModel.updateInteractionInterval(
-                binding.editContact.text.toString(),
-                timeConverter.daysToMillis(binding.editInterval.text.toString().toLong())
+    }
+
+    private fun setupBinding() {
+        if (args.editExisting) {
+            args.mafkirContact.let {
```

Comment:
args.mafkirContact!!.let instead of the it

Reply:
nice dn